docs/user-defined-functions/index.md

```diff
@@ -0,0 +1,577 @@
+
+First, we need to port forward the Flink Job Manager pod so the Flink SQL CLI can access it:
+
+```shell
+kubectl -n flink port-forward <job-manager-pod> 8081:8081
+```
+
+The job manager pod will have the name format `session-cluster-udf-<alphanumeric>`, your `kubectl` should tab-complete the name. 
+If it doesn't then you can find the job manager name by running `kubectl -n flink get pods`.
+
+Next, we will create a container with our JAR mounted into it:
+
+```shell
+podman run -it --rm --net=host \
+    -v ~/currency-converter/target/currency-converter-1.0-SNAPSHOT.jar:/opt/currency-converter-1.0-SNAPSHOT.jar:Z \
+    quay.io/streamshub/flink-sql-runner:0.2.0 \
+        /opt/flink/bin/sql-client.sh embedded
+```
+
+> Note: Flink [ships optional dependencies in `/opt`](https://nightlies.apache.org/flink/flink-docs-release-2.0/docs/dev/configuration/advanced/#anatomy-of-the-flink-distribution), so that's a good place to mount our JAR.
+
+> Note: Don't forget the `:Z` at the end of the volume mount if using a system with SELinux! Otherwise, you will get a permission error when trying to use the JAR later.
+
+Once we're in the Flink SQL CLI, we will first create a table for the generated international sales records:
+
+```sql
+CREATE TABLE InternationalSalesRecordTable ( 
+    invoice_id STRING, 
+    user_id STRING, 
+    product_id STRING, 
+    quantity STRING, 
+    unit_cost STRING, 
+    `purchase_time` TIMESTAMP(3) METADATA FROM 'timestamp', 
+    WATERMARK FOR purchase_time AS purchase_time - INTERVAL '1' SECOND 
+) WITH ( 
+    'connector' = 'kafka',
+    'topic' = 'flink.international.sales.records', 
+    'properties.bootstrap.servers' = 'my-cluster-kafka-bootstrap.flink.svc:9092', 
+    'properties.group.id' = 'international-sales-record-group', 
+    'value.format' = 'avro-confluent', 
+    'value.avro-confluent.url' = 'http://apicurio-registry-service.flink.svc:8080/apis/ccompat/v6', 
+    'scan.startup.mode' = 'latest-offset'
+);
+```
+
+We can do a simple query to verify that the table was created correctly and that the data is flowing (give it a few seconds to start receiving data):
+
+```sql
+SELECT * FROM InternationalSalesRecordTable;
+```
+
+If that worked, we can now register our UDF as a [temporary catalog function](https://nightlies.apache.org/flink/flink-docs-release-2.0/docs/dev/table/sql/create/#create-function):
+
+```sql
+CREATE TEMPORARY FUNCTION currency_convert
+AS 'com.github.example.CurrencyConverter'
+USING JAR '/opt/currency-converter-1.0-SNAPSHOT.jar';
+```
+
+> Note: Temporary catalog functions [only live as long as the current session](https://nightlies.apache.org/flink/flink-docs-release-2.0/docs/dev/table/functions/overview/#types-of-functions). You can omit the `TEMPORARY` keyword to create a catalog function that persists across sessions.
```

Comment:
So this is absolutely correct. However, a catalog is not available by default, so we just need to be a little more explicit. Something like:
```suggestion
> Note: Temporary catalog functions [only live as long as the current session](https://nightlies.apache.org/flink/flink-docs-release-2.0/docs/dev/table/functions/overview/#types-of-functions). Provided you have a [Flink catalog](https://nightlies.apache.org/flink/flink-docs-release-2.0/docs/dev/table/catalogs/#catalogs) deployed and configured, you can omit the `TEMPORARY` keyword to create a function that persists across sessions.
```